exporter/f680/application_sections/supporting_documents/views.py

```
@@ -0,0 +1,80 @@
+    template_name = "f680/supporting_documents/supporting-documents.html"
+
+    def get_context_data(self, pk, **kwargs):
+        return {
+            "application": self.application,
+            "supporting_documents": self.supporting_documents["results"],
+            "back_link_url": reverse(
+                "f680:summary",
+                kwargs={"pk": self.application_id},
+            ),
+        }
+
+
+class SupportingDocumentsAddView(F680FeatureRequiredMixin, F680SupportingDocumentsMixin, FormView):
+    form_class = F680AttachSupportingDocument
+    template_name = "core/form.html"
+    section = "supporting_documents"
+    section_label = "Supporting Documents"
+
+    @expect_status(
+        HTTPStatus.CREATED,
+        "Error creating F680 document",
+        "Unexpected error creating F680 document",
+    )
+    def post_f680_document(self, data):
+        return post_application_supporting_document(self.request, data, self.application_id)
+
+    def get_success_url(self):
+        return reverse(
+            "f680:supporting_documents:add",
+            kwargs={"pk": self.application_id},
+        )
+
+    def form_valid(self, form):
+        data = form.cleaned_data
+        file = data["file"]
+        description = data["description"]
+        payload = {**get_document_data(file), "description": description, "application": self.application_id}
+        self.post_f680_document(payload)
+        self.update_supporting_application_documents()
+        return super().form_valid(form)
+
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+        context_data["back_link_url"] = self.get_success_url()
+        return context_data
+
+    @expect_status(
+        HTTPStatus.OK,
+        "Error updating F680 supporting documents",
+        "Unexpected error updating F680 supporting documents",
+    )
+    def update_supporting_application_documents(self):
```

This function is to update the ID in the supporting documents incase we need to reference them in a future date. 
I did consider an incremental update but thought this is easiest to keep them in sync incase another process deletes the document(s). 

Also I have left the payload intact so it's consistent with the other sections. i.e label, type but technically this isn't needed. 

Also another function this will have to to check and confirm that 'supporting documents' have been completed else we could have to make another API call just to check if we have documents

---

Yep I like this approach.  Get all the documents from the API and then put them in to the JSON in one shot.

---

One other thing we could do as a follow-up is to raise a validation error of some kind on the application submit API endpoint if we find a disparity between `ApplicationDocument` objects and the JSON here.  What do you think?

---

kind of know what you mean but not sure I 100% follow you mean at the end of the submission ?
just trying to figure the scenario where this might be out of sync

---

Reason I suggest application submission is that is the point that the application becomes immutable from an exporter perspective.  So I had it in my head that for things like this where we create an object alongside the JSON, it could be a good time to check that the objects and JSON tally up and fail loudly if not